### DISCUSSION TITLE
Optimal control ressources

### DISCUSSION THREAD

**Discussion by jbcaillau** [control-toolbox/control-toolbox.github.io#26] [Ideas]
[TwoStageOptimalControl](https://github.com/michaelfreiberger/TwoStageOptimalControl.jl)

**Comment by jbcaillau** [comment]
mpopt 
- casadi based, python, collocation, iPopt
- multi-stage possibilities 
https://mpopt.readthedocs.io/en/latest/notebooks/falcon9_to_orbit.html

**Comment by jbcaillau** [comment]
check also https://dionysos-dev.github.io/Dionysos.jl/dev/generated/Path%20planning/

**Comment by jbcaillau** [comment]
@ocots [optcontrol.jl](https://github.com/ai4energy/OptControl.jl) inactive since 2022
see also https://arxiv.org/pdf/2207.13229

**Comment by jbcaillau** [comment]
@ocots CusADi: 
- https://arxiv.org/abs/2408.09662
- https://github.com/se-hwan/cusadi?tab=readme-ov-file

**Comment by ocots** [comment]
@jbcaillau Guix (https://guix.gnu.org) can be used to perform benchmarking on the same machine with bit-level reproducibility. This makes it possible to fairly compare different versions of the same algorithm.

**Comment by jbcaillau** [comment]
@sshin23 see that InfiniteExaModels.jl v1.0 is just out 👍🏽: nice benchmark in https://www.sciencedirect.com/science/article/pii/S2772508125000201

cc: @ocots @amontoison

**Reply by amontoison** [reply]
Hello everyone 👋
I am with @pulsipher in NZ, maybe we should interact more all together. We have doing more overloap than what I was thinking!

**Reply by jbcaillau** [reply]
@amontoison @pulsipher @sshin23 hello NZ people! we'll sure be glad to interact. some overlap is good for (robust) science 🙂

cc @ocots @PierreMartinon @joseph-gergaud

**Comment by jbcaillau** [comment]
@ocots @PierreMartinon @joseph-gergaud 

From Dartmouth college:
- MPCGPU (MPC only): https://github.com/A2R-Lab/MPCGPU (repo inactif)
- GATO (même team et même genre, linéarisation autour traj. nominale + batch): https://github.com/a2r-lab/GATO (actif)

**Comment by jbcaillau** [comment]
@ocots @PierreMartinon @joseph-gergaud from Imperial (Eric K. *et al*), ICLOCS:
www.ee.ic.ac.uk/ICLOCS/Examples.html
